graphql-dgs-codegen-core/src/test/kotlin/com/netflix/graphql/dgs/codegen/CodeGenTest.kt

```diff
@@ -460,6 +459,81 @@ class CodeGenTest {
         assertCompilesJava(dataTypes + interfaces)
     }
 
+    @Test
+    fun generateInterfaceClassWithInterfaceFields() {
+        // schema contains nullable, non-nullable and list types as interface fields  and fields that are
+        // not interfaces
+        val schema = """
+            interface Pet {
+                id: ID!
+	            name: String
+                address: [String!]!
+                mother: Pet!
+                father: Pet
+            	parents: [Pet]
+             }
+            type Dog implements Pet {
+                id: ID!
+	            name: String
+                address: [String!]!
+                mother: Dog!
+                father: Dog
+            	parents: [Dog]
+            }
+            type Bird implements Pet {
+                id: ID!
+	            name: String
+                address: [String!]!
+                mother: Bird!
+                father: Bird
+            	parents: [Bird]
+            }
+        """.trimIndent()
+
+        val (dataTypes, interfaces) = CodeGen(
+            CodeGenConfig(
+                schemas = setOf(schema),
+                packageName = basePackageName,
+            )
+        ).generate() as CodeGenResult
+
+        Truth.assertThat(interfaces[0].toString()).isEqualTo(
```

Comment:
Should we also assert how `Bird` and `Dog` are defined?

Reply:
The scope of the test is to verify that the interface class is generated as expected. The generation of teh data classes is verified by other tests in any case. I could add those here, but it makes the test a bit to long and unwieldy, so chose not to add it.